controllers/storagecluster/external_resources.go

```diff
@@ -248,6 +241,20 @@ func (obj *ocsExternalResources) ensureDeleted(_ *StorageClusterReconciler, _ *o
 	return reconcile.Result{}, nil
 }
 
+// setExternalOCSResourcesData retrieves data from the secret and stores it in the externalOCSResources global variable.
+func (r *StorageClusterReconciler) setExternalOCSResourcesData(instance *ocsv1.StorageCluster) error {
```

Comment:
check if the reconciliation its get updated if we change the JSON

Reply:
yes this func is being called always which means it will get updated.